### DISCUSSION TITLE
Any solutions to test email received content using a stable email address on cypress 10?

### DISCUSSION THREAD

**Discussion by MatiDiez** [cypress-io/cypress#24627] [Questions and Help]
I'm working with Cypress 10, and before this I was using gmail-tester integration to test some stuff but it doesn't work anymore with javascript + cypress 10 and refactoring will be a lot. 

I have found a solution using ethereal email, nodemailer, and a couple of more plugins and it works perfect for initial interactions, like verify my email, change password, etc. Provided I'm working with a new email account everytime. My problem is that I have many test cases that require me to work with a fully populated account from my app so It is not viable to create everything each time. 

So my question is, do you know of any solution, post, plugin, document that could help me figure this one out? 
Essentially I need to always use the same email address, and be able to access the emails on that mailbox. No matter the service, if it is local or remote. Only thing that matter (for now) is that is free cause I didn't get any budget for this one. 

Thank you!

**Comment by portyu9** [comment]
Hi @MatiDiez 

For this use case, I would avoid Gmail entirely and point the application's **test environment SMTP** at a local test mail server such as **Mailpit**.

Mailpit is free/open source, provides SMTP plus an HTTP API, and lets you reuse a stable recipient such as:

```text
existing-user@example.test
```

Your flow can then be:

1. Keep the same fully populated application account.
2. Point test/staging SMTP at Mailpit.
3. Trigger the action in Cypress.
4. Query Mailpit's API for mail addressed to that user.
5. Assert the subject/body/link.
6. Delete or uniquely correlate the message before the next test.

Use a unique value in each generated message when possible — for example an order ID or correlation token — rather than simply taking "the latest email". That makes parallel tests much safer.

This tests the important part, **your application generating the correct email**, without depending on Gmail's UI/API/authentication policies.

If you specifically need to verify real-world delivery through Gmail, that is a different integration test and I would keep it separate from the main Cypress suite.

Please mark this answer as accepted if it helped, thank you.
